src/components/inputs/InlineField.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useState } from 'react';
+import { TextInputProps } from 'react-native';
+import Input from './Input';
+import {
+  Column,
+  Columns,
+  Inset,
+  Text,
+  useTextStyle,
+} from '@rainbow-me/design-system';
+
+export type InlineFieldProps = {
+  defaultValue?: string;
+  label: string;
+  placeholder?: string;
+  inputProps?: Partial<TextInputProps>;
+  onChangeText: (text: string) => void;
+  onEndEditing?: TextInputProps['onEndEditing'];
+  validations?: {
+    allowCharacterRegex?: { match: RegExp };
+    maxLength?: { value: number };
+  };
+  value: string;
+};
+
+export default function InlineField({
+  label,
+  onChangeText,
+  placeholder,
+  inputProps,
+  validations,
+  onEndEditing,
+  value,
+}: InlineFieldProps) {
+  const textSize = 16;
+  const textStyle = useTextStyle({ size: `${textSize}px`, weight: 'bold' });
+
+  const [inputHeight, setInputHeight] = useState(textSize);
+  const handleContentSizeChange = useCallback(
+    ({ nativeEvent }) => {
+      if (inputProps?.multiline) {
+        const contentHeight = nativeEvent.contentSize.height;
+        if (contentHeight > 30) {
+          setInputHeight(nativeEvent.contentSize.height);
+        } else {
+          setInputHeight(textSize);
+        }
+      }
+    },
+    [inputProps?.multiline]
+  );
+
+  const handleChangeText = useCallback(
+    text => {
+      const { allowCharacterRegex } = validations || {};
+      if (!allowCharacterRegex) {
+        onChangeText(text);
+        return;
+      }
+      if (text === '') {
+        onChangeText(text);
+        return;
+      }
+      if (allowCharacterRegex?.match.test(text)) {
+        onChangeText(text);
```

Comment:
we've been using `react-native-text-input-mask` for some number inputs to avoid incorrect number formats, how do you feel about using it here? the difference would be that now the user could type @@ and then we'll remove the last one, the mask would only allow one @

Reply:
I think `react-native-text-input-mask` only supports numbers?

Reply:
Wait, looks like it supports both... I'll have a look into it.